interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/IdValidator.scala

```diff
@@ -0,0 +1,92 @@
+package pl.touk.nussknacker.engine.compile
+
+import cats.data.Validated.{invalid, valid}
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError._
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.graph.node.NodeData
+
+object IdValidator {
+
+  import cats.implicits._
+
+  private val nodeIdIllegalCharacters = Set('.', '"', '\'')
+
+  def validate(process: CanonicalProcess): ValidatedNel[ProcessCompilationError, Unit] = {
+    val scenarioIdValidationResult =
+      validateIdIsNotEmpty(process).andThen { _ =>
```

Comment:
You can accumulate all these errors using `.product` or `.mapN`

Reply:
I'm using andThen to have more fine-grained control over the errors so that on empty id we dont get IsBlank error and when we have a blank value we don't get LeadingSpaces and TrailingSpaces errors. The result is that user gets more precise error message and FE in most cases gets only 1 error to display.

Reply:
Ok, it makes sense.